test/cqlpy/test_name.py

```diff
@@ -107,25 +115,78 @@ def test_table_name_length_500(cql, test_keyspace, cassandra_bug):
 
 # Verifies that a table name of exactly 207 characters is accepted when CDC is enabled.
 def test_table_name_length_eq_scylla_limit_and_cdc_enabled(cql):
-    with new_test_keyspace(
-        cql, "WITH REPLICATION = {'class': 'SimpleStrategy', 'replication_factor': 1}"
-    ) as keyspace:
+    with new_test_keyspace(cql, REPLICATION) as keyspace:
         with new_named_test_table(
             cql,
             keyspace,
-            padded_name(MAX_NAME_LENGTH),
+            padded_name(NAME_MAX_LENGTH),
             SCHEMA,
             extra="with cdc = {'enabled': true}",
         ):
             pass
 
 
+# Verifies that a table name with all allowed characters is accepted.
+def test_table_name_contains_allowed_characters(cql, test_keyspace):
+    name = NAME_ALLOWED_CHARACTERS
+    with new_named_test_table(cql, test_keyspace, name, SCHEMA):
+        pass
+
+
+# Verifies that a table name with a disallowed character is rejected.
+def test_table_name_contains_disallowed_character(cql, test_keyspace):
+    name = "table-name-with-dash-which-is-not-allowed"
+    with pytest.raises(SyntaxException):
+        with new_named_test_table(cql, test_keyspace, name, SCHEMA):
+            pass
+
+
+# Verifies that a table name with a disallowed character in quotes is rejected.
+def test_table_name_contains_disallowed_character_in_quotes(cql, test_keyspace):
+    name = '"table-name-with-dash-which-is-not-allowed"'
+    with pytest.raises(InvalidRequest, match=name):
+        with new_named_test_table(cql, test_keyspace, name, SCHEMA):
+            pass
+
+
+# Verifies that a table name starting with _ is rejected.
+def test_table_name_starts_with_underscore(cql, test_keyspace):
+    name = "_table_name_starting_with_underscore"
+    with pytest.raises(SyntaxException):
+        with new_named_test_table(cql, test_keyspace, name, SCHEMA):
+            pass
+
+
+# Verifies that a table name starting with _ in quotes is accepted.
+def test_table_name_starts_with_underscore_in_quotes(cql, test_keyspace):
+    name = '"_table_name_starting_with_underscore"'
+    with new_named_test_table(cql, test_keyspace, name, SCHEMA):
+        pass
+
+
+# Verifies that table names are case-insensitive when not quoted.
+def test_table_name_is_case_insensitivity_when_not_quoted(cql, test_keyspace):
+    name = "TABLE_NAME_CASE_INSENSITIVE"
```

Comment:
An even safer way to write this test is to create a unique_name() and add to it at the end "a" vs. "A", so the test can't be affected by existing tables. But I guess this way is safe enough, and the same approach that original test you replaced was using.